xcresultparser/JunitXML.swift

```diff
@@ -170,7 +170,7 @@ struct JunitXML {
             let testcase = thisTest.xmlNode(classname: group.nameString)
             if thisTest.isFailed {
                 let summary = failureSummaries.first { summary in
-                    return summary.testCaseName == thisTest.identifier.replacingOccurrences(of: "/", with: ".")
+                    return summary.testCaseName == self.assembledIdentifier(thisTest.identifier)
                 }
                 if let summary = summary {
```

Comment:
I rather propose the following change:
```
            if thisTest.isFailed {
                if let summary = thisTest.failureSummary(in: failureSummaries) {
                    testcase.addChild(summary.failureXML(projectRoot: projectRoot))
                } else {
                    testcase.addChild(failureWithoutSummary)
                }
            }
```
and then have an extension to ActionTestMetadata in this file:
```
private extension ActionTestMetadata {
    func failureSummary(in summaries: [TestFailureIssueSummary]) -> TestFailureIssueSummary? {
        return summaries.first { summary in
            return summary.testCaseName == identifier.replacingOccurrences(of: "/", with: ".") ||
            summary.testCaseName == "-[\(identifier.replacingOccurrences(of: "/", with: " "))]"
        }
    }
}
```
So we would catch both, objective-c and swift syntax, until somebody comes up with a better solution

Reply:
great! That will do for both objective-c and swift